Sources/Hypertext.swift

```diff
@@ -6,18 +6,23 @@
 //  Copyright © 2016 Sahand Nayebaziz. All rights reserved.
 //
 
-public protocol Renderable {
-    func render() -> String
-    func render(startingWithSpacesCount: Int) -> String
```

Comment:
My intuition tells me that 99% of the time, Hypertext will be used just to generate HTML on a server to send back to a browser. And in that case of generating HTML for a server response, indentation doesn't matter. Right? All you need is valid markup. In fact, if it's not indented and formatted, it should be slightly smaller in size, plus the browser's inspector will format it and indent it for you anyway if you're developing the web application as well.

Uses I can think of for generating HTML that's formatted with indentation are maybe you're generating sample HTML code you want to show in code blocks, or maybe somebody writes a template engine and they want to write files to the file system that are pretty and human-readable. So I definitely think we should support it. I think somebody will find use in it.

So with these thoughts in mind, the "render with formatting" is the alternate case. So I would really like to keep two separate render functions in the `Renderable` protocol because, to me, it's nice and there is value in seeing this come up in Xcode: 

<img width="601" alt="screen shot 2016-11-08 at 8 57 06 am" src="https://cloud.githubusercontent.com/assets/909340/20102507/531b00e2-a594-11e6-9c23-cc39232931ab.png">

One simple `render()` that takes no arguments and just works off of whatever is `Renderable`, be it a String, an Int, a `tag`, an array of `tag`s, and anything you conform to `Renderable`. I think that's important. Anybody can use that all day to render, like maybe on a server, and then if you want to render with formatting, there is a second render function where you can (after this PR is merged) pass a number of spaces to indent by and then all the `Renderable`s that are children know that's what you want to do and render themselves in an indented way however they are supposed to.

So can we keep the `render()` and `render(formatting)`?